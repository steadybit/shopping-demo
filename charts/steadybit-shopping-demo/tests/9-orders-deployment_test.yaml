templates:
  - 9-orders-deployment.yml
chart:
  appVersion: v0.0.0
tests:
  - it: manifest should match snapshot
    set:
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot replica count
    set:
      orders:
        replicaCount: 2
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with horizontalPodAutoscaler and podDisruptionBudget
    set:
      orders:
        replicaCount: 2
        podAffinity: true
        podDisruptionBudget: true
    asserts:
      - matchSnapshot: { }