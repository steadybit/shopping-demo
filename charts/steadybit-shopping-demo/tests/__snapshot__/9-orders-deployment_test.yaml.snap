manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-orders
        run: orders
        service-tier: "1"
        tags.datadoghq.com/service: shop-orders
        tags.datadoghq.com/version: 1.0.0
      name: orders
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          run: orders
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-orders
            run: orders
            service-tier: "1"
            tags.datadoghq.com/service: shop-orders
            tags.datadoghq.com/version: 1.0.0
        spec:
          containers:
            - env:
                - name: ACTIVEMQ_BROKER_HOST
                  value: activemq.steadybit-shopping-demo.svc.cluster.local:61613
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
              image: steadybit/orders-go:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8086
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 3
              name: orders
              ports:
                - containerPort: 8086
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8086
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 250Mi
                  memory: 128Mi
          imagePullSecrets:
            - name: regcredinternal
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: orders
      name: orders
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8086
          protocol: TCP
          targetPort: 8086
      selector:
        run: orders
      type: ClusterIP
manifest should match snapshot replica count:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-orders
        run: orders
        service-tier: "1"
        tags.datadoghq.com/service: shop-orders
        tags.datadoghq.com/version: 1.0.0
      name: orders
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: orders
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-orders
            run: orders
            service-tier: "1"
            tags.datadoghq.com/service: shop-orders
            tags.datadoghq.com/version: 1.0.0
        spec:
          containers:
            - env:
                - name: ACTIVEMQ_BROKER_HOST
                  value: activemq.steadybit-shopping-demo.svc.cluster.local:61613
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
              image: steadybit/orders-go:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8086
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 3
              name: orders
              ports:
                - containerPort: 8086
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8086
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 250Mi
                  memory: 128Mi
          imagePullSecrets:
            - name: regcredinternal
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: orders
      name: orders
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8086
          protocol: TCP
          targetPort: 8086
      selector:
        run: orders
      type: ClusterIP
manifest should match snapshot with horizontalPodAutoscaler and podDisruptionBudget:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        domain: shop-orders
        run: orders
        service-tier: "1"
        tags.datadoghq.com/service: shop-orders
        tags.datadoghq.com/version: 1.0.0
      name: orders
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          run: orders
      strategy:
        rollingUpdate:
          maxSurge: 0
          maxUnavailable: 1
        type: RollingUpdate
      template:
        metadata:
          labels:
            domain: shop-orders
            run: orders
            service-tier: "1"
            tags.datadoghq.com/service: shop-orders
            tags.datadoghq.com/version: 1.0.0
        spec:
          affinity:
            podAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: run
                        operator: In
                        values:
                          - orders
                  topologyKey: topology.kubernetes.io/zone
          containers:
            - env:
                - name: ACTIVEMQ_BROKER_HOST
                  value: activemq.steadybit-shopping-demo.svc.cluster.local:61613
                - name: SPRING_PROFILES_ACTIVE
                  value: kubernetes
                - name: DD_SERVICE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/service']
                - name: DD_VERSION
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.labels['tags.datadoghq.com/version']
              image: steadybit/orders-go:develop
              imagePullPolicy: Always
              livenessProbe:
                httpGet:
                  path: /actuator/health/liveness
                  port: 8086
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 3
              name: orders
              ports:
                - containerPort: 8086
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /actuator/health/readiness
                  port: 8086
                initialDelaySeconds: 10
                periodSeconds: 10
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 500m
                  ephemeral-storage: 500Mi
                  memory: 1024Mi
                requests:
                  cpu: 50m
                  ephemeral-storage: 250Mi
                  memory: 128Mi
          imagePullSecrets:
            - name: regcredinternal
          serviceAccountName: steadybit-demo
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        run: orders
      name: orders
      namespace: NAMESPACE
    spec:
      ports:
        - port: 8086
          protocol: TCP
          targetPort: 8086
      selector:
        run: orders
      type: ClusterIP
  3: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      name: orders-pdb
      namespace: NAMESPACE
    spec:
      minAvailable: 1
      selector:
        matchLabels:
          run: orders
